homeassistant/components/sensor/edp_redy.py

```diff
@@ -0,0 +1,122 @@
+"""Support for EDP re:dy sensors."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from homeassistant.components.edp_redy import EdpRedyDevice, EDP_REDY
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['edp_redy']
+
+# Load power in watts (W)
+ATTR_ACTIVE_POWER = 'active_power'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Perform the setup for re:dy devices."""
+    from edp_redy.session import ACTIVE_POWER_ID
+
+    session = hass.data[EDP_REDY]
+    devices = []
+
+    # Create sensors for modules
+    for device_json in session.modules_dict.values():
+        if "HA_POWER_METER" not in device_json["Capabilities"]:
+            continue
+        devices.append(EdpRedyModuleSensor(session, device_json))
+
+    # Create a sensor for global active power
+    devices.append(EdpRedySensor(session, ACTIVE_POWER_ID, "Power Home",
+                                 "mdi:flash", "W"))
+
+    add_devices(devices)
+
+
+class EdpRedySensor(EdpRedyDevice, Entity):
+    """Representation of a EDP re:dy generic sensor."""
+
+    def __init__(self, session, sensor_id, name, icon, unit):
+        """Initialize the sensor."""
+        EdpRedyDevice.__init__(self, session, sensor_id, name)
+
+        self._icon = icon
+        self._unit = unit
+
+        self._update_state()
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this sensor."""
+        return self._unit
+
+    def _data_updated(self):
+        self._update_state()
+        super()._data_updated()
+
+    def _update_state(self):
```

Comment:
Rename this to `async_update` and make it a coroutine. There's no I/O here, correct?